pkg/controller/deployment_util.go

```diff
@@ -435,6 +435,10 @@ func copyMachineDeploymentNodeTemplatesToMachineSet(deployment *v1alpha1.Machine
 
 	isNodeTemplateChanged := !(apiequality.Semantic.DeepEqual(deployment.Spec.Template.Spec.NodeTemplateSpec, is.Spec.Template.Spec.NodeTemplateSpec))
 
+	if isNodeTemplateChanged {
+		glog.V(2).Infof("Observed a change in NodeTemplate of Machine Deployment %s. Changing NodeTemplate from %+v to \n %+v.", deployment.Name, is.Spec.Template.Spec.NodeTemplateSpec, deployment.Spec.Template.Spec.NodeTemplateSpec)
+	}
+
 	if isNodeTemplateChanged {
```

Comment:
Why do you have duplicate `if` checks on line 438 and line 442 consecutively?

Reply:
oops, that's a mistake, thanks .. .. 👍